### ISSUE TITLE
KeyError: 'boto_route53.present' when running Route53 Formula

### ISSUE THREAD

**Issue by rehmanzile** [saltstack/salt#36932]
KeyError: 'boto_route53.present' when running "configure_aws.sls" formula to update Route53 entry.

``` yaml
update route 53:
   boto_route53.present:
    - runas: root
    - name: 'route53-test1.internal.domain.com.'
    - value: '1.1.1.1'
    - zone: 'domain.com.'
    - ttl: 60
    - record_type: A
    - region: us-west-2
    - keyid: {{ salt['pillar.get']('environment:aws:route53:key_id', '') }}
    - key: {{ salt['pillar.get']('environment:aws:route53:key', '') }}
```

However, the following exception is thrown by the minion even though the aws module is installed:

```
minion1-test:
    The minion function caused an exception: Traceback (most recent call last):
      File "/usr/lib/python2.7/dist-packages/salt/minion.py", line 1161, in _thread_return
        return_data = func(*args, **kwargs)
      File "/usr/lib/python2.7/dist-packages/salt/modules/state.py", line 297, in apply_
        return highstate(**kwargs)
      File "/usr/lib/python2.7/dist-packages/salt/modules/state.py", line 523, in highstate
        whitelist=kwargs.get('whitelist')
      File "/usr/lib/python2.7/dist-packages/salt/state.py", line 2968, in call_highstate
        return self.state.call_high(high)
      File "/usr/lib/python2.7/dist-packages/salt/state.py", line 2067, in call_high
        ret = dict(list(disabled.items()) + list(self.call_chunks(chunks).items()))
      File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1623, in call_chunks
        running = self.call_chunk(low, running, chunks)
      File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1769, in call_chunk
        self._mod_init(low)
      File "/usr/lib/python2.7/dist-packages/salt/state.py", line 612, in _mod_init
        self.states['{0}.{1}'.format(low['state'], low['fun'])]  # pylint: disable=W0106
      File "/usr/lib/python2.7/dist-packages/salt/utils/lazy.py", line 90, in __getitem__
        raise KeyError(key)
    KeyError: 'boto_route53.present'
```
### Versions Report

``` text
$ salt --versions-report
                  Salt: 2015.5.3
                Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
                Jinja2: 2.7.2
              M2Crypto: 0.21.1
        msgpack-python: 0.3.0
          msgpack-pure: Not Installed
              pycrypto: 2.6.1
               libnacl: Not Installed
                PyYAML: 3.10
                 ioflo: Not Installed
                 PyZMQ: 14.0.1
                  RAET: Not Installed
                   ZMQ: 4.0.4
                  Mako: Not Installed
               Tornado: Not Installed
 Debian source package: 2015.5.3+ds-1trusty1
```

Minion version is **2016.3.3**.

Both salt master and minion are Ubuntu machines.

**Comment by buildmaster-nyansa** [comment]
+1

**Comment by Ch3LL** [comment]
@rehmanzile what version of boto do you have installed?

If you run these commands in a python interpretter what do you see:

```
    import boto
    import boto.route53
```

**Comment by buildmaster-nyansa** [comment]
@Ch3LL My mistake! Boto was not installed on the minion at all. I couldn't figure it out from the error message. Thanks for your help.

**Comment by Ch3LL** [comment]
@buildmaster-nyansa glad you got it past that error now. For the other issue which version of salt are you on? Also if you run the state with `salt-call -ldebug <state>` that should give us some more data and paste any related information here. Also make sure to sanitize that output. :)

**Comment by buildmaster-nyansa** [comment]
@Ch3LL I am running salt version 2015.5.3. Looks like there was a [fix](https://github.com/saltstack/salt/issues/27374) checked in during October. 

Btw, do I need to copy the <state> file to the minion to run the salt-call command?

**Comment by Ch3LL** [comment]
ahh nice find! I would really recommend upgrading to our latest 2016.3.3 release since 2015.5.3 is end of lifed i believe. If you upgrade are you still seeing the issue?

Also for salt-call you can simply run `salt-call -ldebug <statefile>` and it will make a call to the master for that state file. Its only when you run `salt-call --local` tat you would need the state file on hte minion, because local essentially means masterless and does not attempt to make a call out to the master. In my experience running `salt-call -ldebug` has led to me being able to debug a lot of things because it logs what is occuring on the minion more thoroughly. :) hope that helps

**Comment by buildmaster-nyansa** [comment]
First off, I am getting this error after the upgrade:

```shell
# salt 'staging3-manager01-ui*' state.apply \ saltenv=staging3
Salt request timed out. The master is not responding. If this error persists after verifying the master is up, worker_threads may need to be increased.
```


Also, when I created the machine, salt-cloud also reports an error:

```shell
root@ip-172-31-48-39:/etc/salt# salt-cloud -P -p ec2_west_2xlarge_manager_staging3 staging3-manager01-ui-test
staging3-manager01-ui-test:
    ----------
    Provisioning:
        VM being provisioned in parallel. PID: 13106
[ERROR   ] Public IP not detected.  If private IP is meant for bootstrap you must specify "ssh_interface: private_ips" in your profile.
```

Eventhough profile has the right entry:

```yaml
ec2_west_2xlarge_manager_staging3:
  provider: ec2-us-west-2-manager-private 
  image: ami-xxxxxx
  size: c4.2xlarge 
  spot_config:
    spot_price: 0.15
  ssh_interface: private_ips
  ssh_username: user1 
  tag: {'Environment': 'staging3'}
  sync_after_install: grains
```

**Comment by buildmaster-nyansa** [comment]
Minions were able to attach when I upgraded salt master from Ubuntu 14.04 to 16.04.

```bash
$ salt --versions-report
Salt Version:
           Salt: 2016.3.3
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: 3.5.0
       dateutil: 2.4.2
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.12 (default, Jul  1 2016, 15:12:24)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: Ubuntu 16.04 xenial
        machine: x86_64
        release: 4.4.0-43-generic
         system: Linux
        version: Ubuntu 16.04 xenial
```

**Comment by buildmaster-nyansa** [comment]
This error still exists though:

```bash
# sudo salt-cloud -P -p ec2_west_2xlarge_manager_staging3 staging3-manager01-ui-test
staging3-manager01-ui-test:
    ----------
    Provisioning:
        VM being provisioned in parallel. PID: 2175
[ERROR   ] Public IP not detected.  If private IP is meant for bootstrap you must specify "ssh_interface: private_ips" in your profile.
```

**Comment by Ch3LL** [comment]
@rehmanzile that error is a duplicate of this #34627 which was fixed in #34641  if you try that fix do you still see the error?

**Comment by rehmanzile** [comment]
@Ch3LL The fix was checked in around July. I don't see a release for 2016.7.X? How do I pickup the fix?
